shared/reducers/fs.tsx

```diff
@@ -269,16 +279,39 @@ export default Container.makeReducer<FsGen.Actions, Types.State>(initialState, {
   },
   [FsGen.tlfSyncConfigLoaded]: (draftState, action) => {
     const oldTlfList = draftState.tlfs[action.payload.tlfType]
-    draftState.tlfs[action.payload.tlfType] = new Map([
-      ...oldTlfList,
-      [
-        action.payload.tlfName,
-        {
-          ...(oldTlfList.get(action.payload.tlfName) || Constants.unknownTlf),
-          syncConfig: action.payload.syncConfig,
-        },
-      ],
-    ])
+    const oldTlfFromFavorites = oldTlfList.get(action.payload.tlfName) || Constants.unknownTlf
+    if (oldTlfFromFavorites !== Constants.unknownTlf) {
+      draftState.tlfs[action.payload.tlfType] = new Map([
+        ...oldTlfList,
+        [
+          action.payload.tlfName,
+          {
+            ...oldTlfFromFavorites,
+            syncConfig: action.payload.syncConfig,
+          },
+        ],
+      ])
+      return
+    }
+
+    const tlfPath = Types.pathConcat(
+      Types.pathConcat(Constants.defaultPath, action.payload.tlfType),
+      action.payload.tlfName
+    )
+    const oldTlfFromAdditional = draftState.tlfs.additionalTlfs.get(tlfPath) || Constants.unknownTlf
```

Comment:
I guesss you've already done the work, but it seems odd to me to support syncing non-canonical TLFs.
 (1) it seems like we're encouraging you to keep them around longer
 (2) aren't they often already only existing locally?

Reply:
Sorry for the confusion. Non-canonical TLFs here should really be `Types.Tlf`s with non-preferred names. So `/keybase/private/z,a,jakob223` rather than the "preferred name `/keybase/private/jakob223,a,z`, or `/keybase/private/jakob223@github` rather than `/keybase/private/jakob223`. They are essentially the same TLF but are referred to with different names.

Reply:
Oh, cool - no worries, maybe change the comment a bit.